lib/util/git/index.ts

```diff
@@ -732,12 +734,33 @@ export async function hasDiff(branchName: string): Promise<boolean> {
   }
 }
 
-export async function commitFiles({
-  branchName,
-  files,
-  message,
-  force = false,
-}: CommitFilesConfig): Promise<CommitSha | null> {
+async function pushFiles({ branchName }: PushFilesConfig): Promise<string> {
+  const pushOptions: TaskOptions = {
+    '--force-with-lease': null,
+    '-u': null,
+  };
+  if (getNoVerify().includes('push')) {
+    pushOptions['--no-verify'] = null;
+  }
+
+  const pushRes = await git.push(
+    'origin',
+    `${branchName}:${branchName}`,
+    pushOptions
+  );
+  delete pushRes.repo;
+  logger.debug({ result: pushRes }, 'git push');
+  // Fetch it after create
+  const ref = `refs/heads/${branchName}:refs/remotes/origin/${branchName}`;
+  await git.fetch(['origin', ref, '--force']);
+  return (await git.revparse([branchName])).trim();
+}
+
+export async function commitFiles(
+  commitConfig: CommitFilesConfig,
+  pushCallback: PushCallback | null
```

Comment:
We'll never call `commitFiles` without second argument anyway, and it would be better to require it explicitly from the caller side. The whole callback idea is the trade-off which is made just because importing `platform` directly from the `git` would cause circular dependency. I've tried to solve it, but turns out it requires huuuge refactoring.

Reply:
Did you consider whether to add platform.commitFiles(), which for most platforms should just pass through to git directly, but for GitHub can add extra logic?

Reply:
🤔 Should look fine, though we still need to extract reusable parts from `git.commitFiles`. Originally I was about to deliver reasonably hacky solution faster, but here we go after 3 months

Reply:
Mostly just git.commitFiles with new parameter push=true?

Reply:
Let's try

Reply:
> Did you consider whether to add platform.commitFiles(), which for most platforms should just pass through to git directly, but for GitHub can add extra logic?

Maybe the platform commit function can be optional and we call  the existing git based function then?

I think the GitHub algorithm is:
- commit files via api
- call git layer to fetch and cache branch

Reply:
Our local git tries to `git add`/`git rm` one-by-one to determine which files should be ignored. Don't we need this logic to be executed before platform-native commit?

Reply:
Actually, this filtering mostly was the reason why I've chosen to use callback. I'm not sure how GitHub handles that, and didn't want to check this, aiming for maximum compatibility with the current implementation.

Reply:
@rarkins Can you tell by looking at logs, can we ignore filtering issue or still have to perform it by running `git` locally?
(relevant log messages are `Detected empty commit - aborting git push` and `No file changes detected. Skipping commit`)

Reply:
No, we need to keep that check